### ISSUE TITLE
Unable to get fingerprint vectors

### ISSUE THREAD

**Issue by NadavQuris** [rdkit/rdkit#5686]
Hi, I'm trying to calculate the RDKit fingerprints. There are several chemicals for which I could not do that, and they all yield the same exception.

I'm using this function:

```python
def get_fingerprint(inchi: str):
    molecule = Chem.MolFromInchi(inchi)
    fingerprint = Chem.RDKFingerprint(molecule)
    fingerprint = fingerprint.ToList()
    return fingerprint
```

And for some chemicals I get, in the `ToList` line, the exception `IndexError: cannot fit 'int' into an index-sized integer`.

An example for such a chemical is gold trichloride, which has the inchi `InChI=1S/Au.3ClH/h;3*1H/q+3;;;/p-3` and the smiles `[Au+3].[Cl-].[Cl-].[Cl-]`.

I'm using rdkit version 2022.3.5 on macOS 12.1. I'm using conda and install with `pip install rdkit`.

**Comment by greglandrum** [comment]
Please provide the specific input which is causing the problem so that we can try to help you here

**Comment by NadavQuris** [comment]
I did, see the above inchi (and its corresponding smiles string) above. Do you need something more than this?

**Comment by bp-kelley** [comment]
I believe this is the same bug as #5677

Small reproducible:

```
>>> m = Chem.MolFromSmiles("[Cl-]")
>>> fp = Chem.RDKFingerprint(m)
>>> fp.ToList()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: cannot fit 'int' into an index-sized integer
>>> 
```

Interestingly, this works:

```
>>> list(fp)
[0,0,.......]
```

**Comment by greglandrum** [comment]
This should have been fixed with the fix for #5677, which was just merged and which will be in the 2022.09.2 patch release.

Please reopen if that's not the case.
